src/NzbDrone.Core/Validation/RuleBuilderExtensions.cs

```diff
@@ -38,6 +38,10 @@ public static class RuleBuilderExtensions
         {
             return ruleBuilder.SetValidator(new RegularExpressionValidator(@"^(?!\/?https?://[-_a-z0-9.]+)", RegexOptions.IgnoreCase)).WithMessage($"Must be a valid URL path (ie: '{example}')");
         }
+        public static IRuleBuilderOptions<T, string> ActuallyValidUrlBase<T>(this IRuleBuilder<T, string> ruleBuilder, string example = "sonarr/base")
```

Comment:
Why do we need another validator? Why is this one actually valid and the other isn't? The other requires a leading slash which is valid and already in use for download clients.

Reply:
The 'Actually' part is there to catch attention in case I forget to mention that the original Validator is buggy.

Saying that anything is a valid url base, that is not a root url is a faulty logic by itself. If we include the fact that the root url regex itself is buggy, we raise the bug potential to even higher level.

![image](https://user-images.githubusercontent.com/12481929/144916297-b6178533-7d87-49da-9cf9-e8b13a21a60a.png)
![image](https://user-images.githubusercontent.com/12481929/144916357-80549284-ff0b-4197-a055-26303a50eec6.png)
Both picture above show invalid root urls, which are being accepted by the current validator. 

I've decided to first create a new good validator, before wandering into other components, to retro fit the old urls with database migration.

Reply:
Other characters are valid, they just need to be URL encoded though.

> I've decided to first create a new good validator

I think we can just fix the one that already exists and if there are issues we fix them, though tests should be added.

> before wandering into other components, to retro fit the old urls with database migration.

Presumably these URL bases are already working, so I don't think there is anything to migrate.

Reply:
> Other characters are valid, they just need to be URL encoded though.

This implies that the content in the Url Base is *before* pct encode, which means anything could be added there, as they will be encoded properly.

However, if we say that here must be *after* pct encode the path must be match the [specification](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3).

This whole sort of unspecified state is why I rather didn't touch the older validations, as it might be totally possible that the user put something there, that works *by magic* and the validator would complain about it after upgrade.

Reply:
Then I will just implement Url Base to comply with rfc3986 paths and if it cases validation errors upon edits, users will have to manually migrate it for themselves? That's acceptable for me.

Reply:
Yeah, I think that's safe enough.